### PR TITLE
meteor: Add new async methods from 2.9

### PR DESCRIPTION
**Pull request by ebroder**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.meteor.com/changelog.html#v2920221212
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by radekmie — COMMENTED** [review]
Please file this PR against the Meteor Core repository instead - the typings are there now. If you have problem using them, please let us know, so we could fix that.

**Comment by ebroder** [comment]
Hmm. I filed an issue against `zodern:types` a few months ago (https://github.com/zodern/meteor-types/issues/5) and still haven't gotten any resolution, so I've been unable to adopt the types from Meteor core. Should I have filed the issue against Meteor instead?

**Comment by MaxHershey** [comment]
I too struggle to get zodern:types working properly, also I've been scouring for types for the new meteor accounts-2fa types. Could these be added to @types/meteor as well?

**Comment by radekmie** [comment]
OK, then let's do it like this: I'll review this PR later today, but please make sure that the types in Core repo first.

File: types/meteor/accounts-base.d.ts [hunk 1]
```diff
@@ -31,6 +33,16 @@ declare module 'meteor/accounts-base' {
             callback?: (error?: Error | Meteor.Error | Meteor.TypedError) => void,
         ): string;
 
+        function createUserAsync(
+            options: {
+                username?: string | undefined;
+                email?: string | undefined;
+                password?: string | undefined;
+                profile?: Object | undefined;
```
Comment (radekmie) [inline types/meteor/accounts-base.d.ts:41]: Both `Object` and `object` aren't restrictive enough. `Record<string, unknown>` would be safer here.

**Review by radekmie — CHANGES_REQUESTED** [review]

File: types/meteor/accounts-base.d.ts (hunk 1 above)
Comment (ebroder) [reply types/meteor/accounts-base.d.ts:41]: Hmm. This is just copy-pasted from the definition of `createUser` above. Can we merge this as-is in the interests of keeping those two in sync, and keeping this in sync with the Meteor core definitions, and I'll follow up with another PR to fix the uses of `Object` across the board in the type definitions?

Comment (radekmie) [reply types/meteor/accounts-base.d.ts:41]: OK, sure.

**Review by radekmie — APPROVED** [review]

**Comment by ebroder** [comment]
Ready to merge

**Pull request merged**